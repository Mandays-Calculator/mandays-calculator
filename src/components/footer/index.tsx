export { default as Footer } from "./Footer";
export { StyledFooterContainer, StyledFooterText } from "./styles";
