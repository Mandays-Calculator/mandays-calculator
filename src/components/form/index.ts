export * from "./checkbox";
export * from "./checkbox-group";
export * from "./text-field";
export * from "./date-picker";
