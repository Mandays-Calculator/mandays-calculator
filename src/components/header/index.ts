export {
  StyledAppBar,
  StyledToolBarContainer,
  StyledSearchSelect,
  popperStyle,
  StyledProjectTitle,
} from "./styles";
