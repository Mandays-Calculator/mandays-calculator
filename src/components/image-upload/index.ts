export { default as ImageUpload } from "./ImageUpload";
export * from "./styles";
