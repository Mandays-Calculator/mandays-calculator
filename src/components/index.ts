export * from "./form/text-field";
export * from "./form/Form";
export * from "./title";
