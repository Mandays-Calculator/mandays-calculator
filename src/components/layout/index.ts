export { default as Layout } from "./Layout";
export {
  StyledLayoutContainer,
  StyledLayoutContent,
  ContentContainer,
} from "./styles";
