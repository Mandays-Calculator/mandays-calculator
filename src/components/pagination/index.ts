export { default as Pagination } from "./Pagination";
export * from "./styles";
