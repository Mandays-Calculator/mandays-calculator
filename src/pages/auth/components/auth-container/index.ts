export { default as AuthContainer } from "./AuthContainer";
export {
  StyledAuthContainer,
  StyledContainer,
  StyledLogoContainer,
  StyledImageContainer,
  StyledLabel,
  StyledTitle,
  StyledGrid,
  AuthImageContainer,
} from "./styles";
