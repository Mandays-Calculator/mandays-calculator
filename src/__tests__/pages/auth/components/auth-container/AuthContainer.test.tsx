import { ReactNode } from 'react';

import { RenderResult, render } from '@testing-library/react';

import { AuthContainer } from '~/pages/auth/components/auth-container';
import LoginIcon from "~/assets/svg/login-icon.svg";

describe('GIVEN AuthContainer Component', () => {
    test('WHEN AuthContainer Component is called with default props, THEN it should render correctly', () => {
        const { asFragment } = render(<AuthContainer />);

        expect(asFragment()).toMatchSnapshot(`component-default-props`);
    });

    test('WHEN AuthContainer Component is called with custom children, THEN it should render correctly', () => {
        const customChildren = <div>Custom Content</div>;
        const { asFragment } = renderAuthContainer(customChildren);

        expect(asFragment()).toMatchSnapshot(`component-custom-children`);
    });

    test('WHEN AuthContainer Component is called with different children (image), THEN it should render correctly', () => {
        const imageChildren = (
            <img src={LoginIcon} alt="example" width="100%" height="auto" />
        );
        const { asFragment } = renderAuthContainer(imageChildren);

        expect(asFragment()).toMatchSnapshot(`component-image-children`);
    });

    test('WHEN AuthContainer Component is called with different children (nested components), THEN it should render correctly', () => {
        const nestedChildren = (
            <div>
                <h1>Nested Content</h1>
                <p>Some text inside nested content.</p>
            </div>
        );
        const { asFragment } = renderAuthContainer(nestedChildren);

        expect(asFragment()).toMatchSnapshot(`component-nested-components`);
    });

    test('WHEN AuthContainer Component is called with empty children, THEN it should render correctly', () => {
        const { asFragment } = renderAuthContainer(null);

        expect(asFragment()).toMatchSnapshot(`component-empty-children`);
    });

    test('WHEN AuthContainer Component is called with multiple types of children, THEN it should render correctly', () => {
        const mixedChildren = (
            <>
                <div>First Child</div>
                <img src={LoginIcon} alt="example" width="100%" height="auto" />
                <p>Text Child</p>
            </>
        );
        const { asFragment } = renderAuthContainer(mixedChildren);

        expect(asFragment()).toMatchSnapshot(`component-multiple-types`);
    });
});

const renderAuthContainer = (children: ReactNode): RenderResult => {
    return render(
        <AuthContainer>{children}</AuthContainer>
    );
}