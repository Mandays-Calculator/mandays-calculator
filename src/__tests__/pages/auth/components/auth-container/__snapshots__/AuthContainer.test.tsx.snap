// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN AuthContainer Component WHEN AuthContainer Component is called with custom children, THEN it should render correctly: component-custom-children 1`] = `
<DocumentFragment>
  <div
    class="css-16a6vc1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1fu5yrq-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-go2vvz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3.75 css-8w0tzz-MuiGrid-root"
        >
          <div
            class="css-x7y3mw"
          >
            <img
              alt="login"
              height="auto"
              src="[object Object]"
              width="100%"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k59ms1-MuiGrid-root"
        >
          <div>
            Custom Content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN AuthContainer Component WHEN AuthContainer Component is called with default props, THEN it should render correctly: component-default-props 1`] = `
<DocumentFragment>
  <div
    class="css-16a6vc1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1fu5yrq-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-go2vvz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3.75 css-8w0tzz-MuiGrid-root"
        >
          <div
            class="css-x7y3mw"
          >
            <img
              alt="login"
              height="auto"
              src="[object Object]"
              width="100%"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k59ms1-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN AuthContainer Component WHEN AuthContainer Component is called with different children (image), THEN it should render correctly: component-image-children 1`] = `
<DocumentFragment>
  <div
    class="css-16a6vc1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1fu5yrq-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-go2vvz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3.75 css-8w0tzz-MuiGrid-root"
        >
          <div
            class="css-x7y3mw"
          >
            <img
              alt="login"
              height="auto"
              src="[object Object]"
              width="100%"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k59ms1-MuiGrid-root"
        >
          <img
            alt="example"
            height="auto"
            src="[object Object]"
            width="100%"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN AuthContainer Component WHEN AuthContainer Component is called with different children (nested components), THEN it should render correctly: component-nested-components 1`] = `
<DocumentFragment>
  <div
    class="css-16a6vc1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1fu5yrq-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-go2vvz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3.75 css-8w0tzz-MuiGrid-root"
        >
          <div
            class="css-x7y3mw"
          >
            <img
              alt="login"
              height="auto"
              src="[object Object]"
              width="100%"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k59ms1-MuiGrid-root"
        >
          <div>
            <h1>
              Nested Content
            </h1>
            <p>
              Some text inside nested content.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN AuthContainer Component WHEN AuthContainer Component is called with empty children, THEN it should render correctly: component-empty-children 1`] = `
<DocumentFragment>
  <div
    class="css-16a6vc1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1fu5yrq-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-go2vvz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3.75 css-8w0tzz-MuiGrid-root"
        >
          <div
            class="css-x7y3mw"
          >
            <img
              alt="login"
              height="auto"
              src="[object Object]"
              width="100%"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k59ms1-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN AuthContainer Component WHEN AuthContainer Component is called with multiple types of children, THEN it should render correctly: component-multiple-types 1`] = `
<DocumentFragment>
  <div
    class="css-16a6vc1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1fu5yrq-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-go2vvz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3.75 css-8w0tzz-MuiGrid-root"
        >
          <div
            class="css-x7y3mw"
          >
            <img
              alt="login"
              height="auto"
              src="[object Object]"
              width="100%"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k59ms1-MuiGrid-root"
        >
          <div>
            First Child
          </div>
          <img
            alt="example"
            height="auto"
            src="[object Object]"
            width="100%"
          />
          <p>
            Text Child
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
